File: examples/tools/Wires/inflate.cpp
#include <cstdlib>
#include <iostream>
#include <string>

#include <Core/EigenTypedef.h>
#include <IO/MeshWriter.h>
#include <Math/MatrixUtils.h>
#include <Wires/WireNetwork/WireNetwork.h>
#include <Wires/Inflator/InflatorEngine.h>

void print_usage(char* prog_name) {
    std::cout << "Usage: " << prog_name
        << " thickness"
        << " in_wire_file"
        <<" out_mesh_file"
        << std::endl;
}

void write_mesh(const std::string& filename,
        const MatrixFr& vertices, const MatrixIr& faces) {
    VectorI voxels = VectorI::Zero(0);
    MeshWriter::Ptr writer = MeshWriter::create(filename);
    writer->write(
            MatrixUtils::flatten<VectorF>(vertices),
            MatrixUtils::flatten<VectorI>(faces),
            voxels,
            vertices.cols(),
            faces.cols(),
            0);
}

int main(int argc, char** argv) {
    if (argc != 4) {
        print_usage(argv[0]);
        return 0;
    }

    Float thickness = atof(argv[1]);

    WireNetwork::Ptr wire_network = WireNetwork::create(argv[2]);
    InflatorEngine::Ptr inflator = InflatorEngine::create("simple", wire_network);
    inflator->set_uniform_thickness(thickness);
    inflator->inflate();

    MatrixFr vertices = inflator->get_vertices();
    MatrixIr faces = inflator->get_faces();

    write_mesh(argv[3], vertices, faces);

    return 0;
}

File: examples/tools/Wires/inflate.py
#!/usr/bin/env python
import argparse
import numpy as np

from pymesh import mesh_io
from PyWires import WireNetwork, InflatorEngine

def parse_args():
    parser = argparse.ArgumentParser(description="Inflate wires");
    parser.add_argument("--thickness", "-t", type=float, help="wire thickness");
    parser.add_argument("wire_file", help="input wire file");
    parser.add_argument("mesh_file", help="output mesh file");
    return parser.parse_args();

def main():
    args = parse_args();

    wire_network = WireNetwork.create(args.wire_file);
    inflator = InflatorEngine.create("simple", wire_network);
    inflator.set_uniform_thickness(args.thickness);
    inflator.inflate();

    mesh_io.save_mesh_raw(
            args.mesh_file,
            inflator.get_vertices(),
            inflator.get_faces());

if __name__ == "__main__":
    main();
